cli: fix get test by uid

jtluka opened the pull request:
This patch fixes retrieval of test by UID. Currently if a user specifies
non numeric id of a test the cli reports failure that id is not numeric.

Additionally I've changed the return code if test was not found. If it was
not found the cli tool should not return non-zero value because non-existing
test is not an error of application, it's just a fact.

Signed-off-by: Jan Tluka jtluka@redhat.com

olichtne commented on perfrepo-cli:
```diff
@@ -157,18 +157,18 @@ class TestCLI(GenericCLI):
             except ValueError:
-                print("ID needs to be an integer!", file=sys.stderr)
-                return -1
+                test = self._perf_api.test_get_by_uid(argv[1])
```
So we're only searching by UID if the value is not int? I think we should always try to search by UID if the search by ID fails (for whatever reason - value error or no test found...).

jtluka replied:
I think this is nonsense. Despite that it is possible to create a test with numeric UID in PerfRepo (just have tested it), users should not do this. Because with every test created the ID increments. Imagine you have test with UID=100. And perfrepo also contains a test with ID=100. Which one it should return? I think that we should really distinguish between numeric and non numeric values specified.

olichtne commented on perfrepo-cli:
```diff
@@ -157,18 +157,18 @@ class TestCLI(GenericCLI):
+
+            if test is None:
+                print("Test not found.")
+                return 0
```
I also disagree with this... we've discussed this in person, it should be a negative value so that scripts can easily detect by the return value if the test was or wasn't found... but the help message shouldn't be printed in such a case.

jtluka replied:
OK, my proposal here is to have two different error return codes. rc=-1 if data could not be retrieved from perfrepo (not an user input error), rc=-2 for anything that is user input error that will print the help.

jtluka committed:
perfrepo-cli: use various error codes to distinguish errors

Currently any non-zero code returned from a command will
print the usage help message. This is not needed when the
error is caused by e.g. test was not found.

The patch adds several error codes to have better granularity
while reporting problem. The usage message will be printed out
only if the syntax is wrong.

Signed-off-by: Jan Tluka <jtluka@redhat.com>

jtluka committed:
cli: fix and enhancements to test get command

This patch fixes retrieval of test by UID. Currently if a user specifies
non numeric id of a test the cli reports failure that id is not numeric.

Additionally user now can query a test explicitly by id or uid. That
allows to search for test uid by a numeric value. Without it the script
would search by id only and exit. Example:

perfrepo-cli test get 21 # searches for test with id=21

If test with id=21 does not exist the script will try to search also for
test with uid=21.

To explicitly search for a test with id or uid, user can do:

perfrepo-cli test get id 21 # searches for test with id=21 only
perfrepo-cli test get uid 21 # searches for test with uid=21 only
perfrepo-cli test get id test1 # this would fail since id must be a number

Signed-off-by: Jan Tluka <jtluka@redhat.com>
